cmd/prompt.go

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	prompt "github.com/c-bata/go-prompt"
+	"github.com/kris-nova/kubicorn/cutil/logger"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var commandFlagSuggestions = make(map[string][]prompt.Suggest)
+var commandMap = make(map[string][]prompt.Suggest)
+var promptSuggestions = []prompt.Suggest{}
+var promptCmd = &cobra.Command{
+	Use:   "prompt",
+	Short: "Open a prompt with auto-completion (non-Windows)",
+	Long: `Use this command to use the Kubicron API via a shell prompt.
+
+This command will open a prompt using go-prompt (with auto-completion) to
+allow you to run commands interactively from the shell.
+Currently this doesn't work on Windows systems`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) > 0 {
+			logger.Critical("Too many arguments.")
+			os.Exit(1)
+		}
+		err := RunPrompt()
+		if err != nil {
+			logger.Critical(err.Error())
+			os.Exit(1)
+		}
+	},
+}
+
+func RunPrompt() error {
+	fmt.Println("Please input your kubicorn commands.")
+	defer fmt.Println("Bye!")
+
+	p := prompt.New(
+		executor,
+		completer,
+		prompt.OptionTitle("kubicorn: interactive kubicorn client"),
+		prompt.OptionPrefix(">> "),
+		prompt.OptionInputTextColor(prompt.Yellow))
+
+	p.Run()
+	return nil
+}
+
+func executor(s string) {
+	s = strings.TrimSpace(s)
+	if s == "" {
+		return
+	} else if s == "quit" || s == "exit" {
+		fmt.Println("Bye!")
+		os.Exit(0)
+		return
+	}
+
+	cmd := exec.Command("/bin/sh", "-c", "kubicorn "+s)
```

Comment:
Not sure is this a needed change, but let's rename this to prevent sort of unwanted effects, as the package is called `cmd` as well.

Reply:
This is required because once inside the prompt, this is the command that will get executed whenever the user pressed enter. Have renamed the variable to remove confusion here.